uint8 state
float32 distance
float32 angle
